Fix AspectJ compiler core dumps, clean up Maven settings

kriegaex opened the pull request:
See commit comments. Especially 40946b420aee0aebbb2f51b57b73ce8e2fabe881 should be interesting, because it fixes your problem.

kriegaex committed:
Ignore AspectJ core dumps

kriegaex committed:
Surefire, Failsafe are plugins, not dependencies

kriegaex committed:
Remove dependencies already declared in parent POM

kriegaex committed:
Use latest AspectJ Maven and fix complianceLevel

complianceLevel should be ${maven.compiler.target}, not ${java.version}.
The latter is undefined in the POM and would fall back to the JDK
version running Maven, e.g. Java 20, which is not supported by AspectJ
1.9.19 yet.

kriegaex committed:
Fix AspectJ compiler core dumps

The problem was that falsely the Allure libraries already declared as
Maven module dependencies and used as AspectJ Maven weaveDependencies
were also declared as plugin dependencies, which is simply wrong.

kriegaex committed:
Ignore all IntelliJ IDEA files

afathonih commented:
I run your PR locally. I can still see 11 occurrences out of 50 runs.
I'm setting up GH Actions so we can reproduce it on the PR as well.

kriegaex commented:
Thanks for checking. I can take a look, if it is reproducible. Please keep me posted.

afathonih commented:
@kriegaex I managed to setup GH action and run your PR there: https://github.com/afathonih/ajcwovenissue/actions/runs/5372836175/jobs/9746611803?pr=1

It's still reproducible.

kriegaex committed:
Clean up AspectJ Maven config some more

according to https://github.com/eclipse-aspectj/aspectj/issues/245#issuecomment-1607501065

afathonih commented:
Thanks for the update. I still need to update my GH Actions-fu here. 
The build [failed](https://github.com/afathonih/ajcwovenissue/actions/runs/5383795928/jobs/9770972271) because it was unable to resolve allure packages.
```
Error:  /home/runner/work/ajcwovenissue/ajcwovenissue/app/target/generated-sources/ksp/com/app/Animal0Factory.kt: (3, 8) Unresolved reference: io
Error:  /home/runner/work/ajcwovenissue/ajcwovenissue/app/target/generated-sources/ksp/com/app/Animal0Factory.kt: (9, 2) Unresolved reference: Step
```

kriegaex commented:
What happens if you run my code locally? I mean, exactly the code from my PR, nothing that might be different in your own main branch. Can I help you any further? For me, it works locally, no matter if I run it once or 25x in a loop.

afathonih commented:
@kriegaex 
I did checkout out your code and run it. The issue was still reproducible.
The GH action also perform a clean (isolated in container) build from each PR. The issue was still reproducible.

On a side note, I tried updating Allure library to a more recent version (2.22.2). It is still failing in GH [build](https://github.com/afathonih/ajcwovenissue/actions/runs/5397993404/jobs/9803281486?pr=2). However, it does not trigger the issue as much as the last one in my machine.

So it's quite strange that it's working on your machine. Maybe there's external factor in play? May some OS specific libraries? Can you try building this repo this inside a virtual machine / container?

By the way, your latest change does not compile:
```
(base) ➜  Github git clone git@github.com:kriegaex/AJ_GH-245.git
Cloning into 'AJ_GH-245'...
remote: Enumerating objects: 67, done.
remote: Counting objects: 100% (67/67), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 67 (delta 21), reused 62 (delta 18), pack-reused 0
Receiving objects: 100% (67/67), 11.28 KiB | 3.76 MiB/s, done.
Resolving deltas: 100% (21/21), done.
(base) ➜  Github cd AJ_GH-245
(base) ➜  AJ_GH-245 git:(main) for i in {1..50}; do echo "Command no. $i"; mvn -Dmaven.test.skip.exec=true package; done
Command no. 1
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] consoleApp                                                         [pom]
[INFO] processor                                                          [jar]
[INFO] app                                                                [jar]
[INFO]
[INFO] ------------------------< org.example:ajcwoven >------------------------
[INFO] Building consoleApp 1.0-SNAPSHOT                                   [1/3]
[INFO]   from pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- kotlin:1.8.10:compile (compile) @ ajcwoven ---
[INFO]
[INFO] --- kotlin:1.8.10:test-compile (test-compile) @ ajcwoven ---
[WARNING] No sources found skipping Kotlin compile
[INFO]
[INFO] -----------------------< org.example:processor >------------------------
[INFO] Building processor 1.0-SNAPSHOT                                    [2/3]
[INFO]   from processor/pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- resources:3.3.0:resources (default-resources) @ processor ---
[INFO] Copying 1 resource
[INFO]
[INFO] --- compiler:3.10.1:compile (default-compile) @ processor ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- kotlin:1.8.10:compile (compile) @ processor ---
[INFO]
[INFO] --- resources:3.3.0:testResources (default-testResources) @ processor ---
[INFO] skip non existing resourceDirectory /Users/akhmad.fathonih/Github/AJ_GH-245/processor/src/test/resources
[INFO]
[INFO] --- compiler:3.10.1:testCompile (default-testCompile) @ processor ---
[INFO] No sources to compile
[INFO]
[INFO] --- kotlin:1.8.10:test-compile (test-compile) @ processor ---
[WARNING] No sources found skipping Kotlin compile
[INFO]
[INFO] --- surefire:2.22.2:test (default-test) @ processor ---
[INFO] Tests are skipped.
[INFO]
[INFO] --- jar:3.3.0:jar (default-jar) @ processor ---
[INFO] Building jar: /Users/akhmad.fathonih/Github/AJ_GH-245/processor/target/processor-1.0-SNAPSHOT.jar
[INFO]
[INFO] --------------------------< org.example:app >---------------------------
[INFO] Building app 1.0-SNAPSHOT                                          [3/3]
[INFO]   from app/pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- resources:3.3.0:resources (default-resources) @ app ---
[INFO] skip non existing resourceDirectory /Users/akhmad.fathonih/Github/AJ_GH-245/app/src/main/resources
[INFO]
[INFO] --- compiler:3.10.1:compile (default-compile) @ app ---
[INFO] Changes detected - recompiling the module!
[INFO]
[INFO] --- kotlin:1.8.10:compile (compile) @ app ---
[INFO] Applied plugin: 'ksp'
[INFO] [ksp] loaded provider(s): [com.processor.BuilderProcessorProvider]
[ERROR] /Users/akhmad.fathonih/Github/AJ_GH-245/app/target/generated-sources/ksp/com/app/Animal0Factory.kt: (3, 8) Unresolved reference: io
[ERROR] /Users/akhmad.fathonih/Github/AJ_GH-245/app/target/generated-sources/ksp/com/app/Animal0Factory.kt: (9, 2) Unresolved reference: Step
```

kriegaex committed:
Fix build failing due to allure-java-commons

allure-java-commons is referenced in generated source code, i.e. it must
not be test-scoped.

kriegaex commented:
Sorry, I pushed another commit to fix the build. It may have worked locally for me, because I copied your `mvn package` instead of using my usual `mvn clean package` or so. Please try again.

kriegaex commented:
Looking good. When searching the [build log](https://github.com/afathonih/ajcwovenissue/actions/runs/5398387880/jobs/9805697330?pr=1) for

![image](https://github.com/afathonih/ajcwovenissue/assets/1537384/925523d3-db08-4447-af15-851abd3d757b)

I find 25x

```text
[INFO] BUILD SUCCESS
```
